aiida/restapi/translator/base.py

```diff
@@ -447,18 +493,34 @@ def get_results(self):
         data = self.get_formatted_result(self._result_type)
         return data
 
-    def _check_pk_validity(self, pk):
+    def _check_id_validity(self, id):
         """
-        Checks whether a pk corresponds to an object of the expected type,
+        Checks whether a id corresponds to an object of the expected type,
         whenever type is a valid column of the database (ex. for nodes,
-        but not for users)_
-        :param pk: (integer) ok to check
-        :return: True or False
+        but not for users)
+        :param id: (integer) ok to check
+        :return: (True/False) if id valid (invalid). If True sets the
+        id filter attribute correctly
         """
         # The logic could be to load the node or to use querybuilder. Let's
         # do with qb for consistency, although it would be easier to do it
         # with load_node
 
+        import uuid
+
+        # Distinguish id and uuid case
+        if type(id) == int:
+            filter =  {
+                    'id': {'==': id}
+                }
+        elif type(id) == uuid.UUID:
```

Comment:
Re the discussion before, here I would check that it is a string and use the same logic of load_node.
Actually, is it possible to use load_node rather than manually the querybuilder?
Or maybe decouple the filters from the logic of load_node, so it can be reused both in load_node and here?